datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DiscardDescriptionModal.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { Modal, Button } from 'antd';
+import { useRouteToTab } from '../../../EntityContext';
+
+type Props = {
+    buttonProps?: Record<string, unknown>;
+    isDescriptionUpdated?: boolean;
+    urn: string;
+};
+
+export const DiscardDescriptionModal = ({ buttonProps, isDescriptionUpdated, urn }: Props) => {
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const routeToTab = useRouteToTab();
+    const localStorageDictionary = localStorage.getItem('editedDescriptions');
```

Comment:
This component should not need to read from local storage. The parent component should decide whether it should be shown or not.

Reply:
Modified the DiscardDescriptionModal.ts.

import React from 'react';
import { Modal, Button } from 'antd';

type Props = {
    visible?: boolean;
    onDiscard?: () => void;
    onCancel?: () => void;
};

export const DiscardDescriptionModal = ({ visible, onDiscard, onCancel }: Props) => {
    return (
        <>
            <Modal
                title="Discard Changes"
                visible={visible}
                destroyOnClose
                onCancel={onCancel}
                footer={[
                    <Button type="text" onClick={onCancel}>
                        Cancel
                    </Button>,
                    <Button onClick={onDiscard}>Discard</Button>,
                ]}
            >
                <p>Changes will not be saved. Do you want to proceed?</p>
            </Modal>
        </>
    );
};